lib/backends/sdc/containers.js

```diff
@@ -559,60 +525,302 @@ function addRulesToFWAPI(opts, rules, payload, callback) {
+     *    - 'bridge' or no network specified: default fabric and...
+     *      - if publishing ports or building: also external network
+     *    - fabric name specified: named fabric only
+     *    unsupported:
+     *
+     * We need to determine our network 'mode'. Docker defines the following
+     * networking modes:
+     *
+     * 1. 'legacy bridge' - the default currently supported.
+     * 2. 'host' - not meaningful for us
+     * 2. 'container' - like 'host', not meaningful
+     * 3. 'none' - eventually supportable
+     * 4. 'user-defined bridge' - probably supportable
+     * 5. 'user-defined overlay' - definitely supportable
+     *
+     */
+    if (!opts.config.overlay.enabled) {
+        return externalNetworkByName(opts, container, payload, callback);
+    } else {
+        if (container.HostConfig.NetworkMode === 'bridge'
+            || container.HostConfig.NetworkMode === 'default'
+            || common.objEmpty(container.HostConfig.NetworkMode)) {
+            return defaultFabricNetwork(opts, container, payload, callback);
+        } else {
+            return namedNetwork(opts, container, payload, callback);
+        }
     }
+}
+
+/*
+ * When fabrics are not configured, we will use the named
+ * network, defaulting to 'external'.
+ */
+function externalNetworkByName(opts, container, payload, callback) {
+    assert.object(opts.config, 'opts.config');
+    assert.optionalString(opts.config.externalNetwork,
+        'opts.config.externalNetwork');
+    assert.object(opts.account, 'opts.account');
+    assert.string(opts.account.uuid, 'opts.account.uuid');
 
     var log = opts.log;
-    log.debug('Fabrics not configured: using external network');
+    var listParams = {
+        name: opts.config.externalNetwork || 'external',
+        fabric: false,
+        provisionable_by: opts.account.uuid
+    };
 
-    // No fabrics configured - fall back to provisioning on the external
-    // network
+    log.debug({ listParams: listParams },
+        format('Networks: fabrics not configured, using network %s',
+        listParams.name));
 
-    var netName = opts.config.externalNetwork || 'external';
-    listNetworks(opts, {name: netName}, function (err, networks) {
-        var external_net;
-        log.debug({err: err, res: networks}, 'list external networks');
+    listNetworks(opts, listParams, function (err, networks) {
+        log.debug({ err: err, res: networks },
+            format('Networks: listNetworks result for %s', listParams.name));
 
         if (err) {
-            callback(errors.napiErrorWrap(err, 'problem listing networks'));
+            callback(errors.napiErrorWrap(err,
+                format('Networks: problem listing network %s',
+                    listParams.name)));
+            return;
+        }
+
+        if (networks.length < 1) {
+            callback(new errors.DockerError(
+                format('unable to find %s network uuid', listParams.name)));
```

Comment:
nit: is `listParams.name` a network _uuid_, or can it be either a name or a uuid? In that case do we want to not specify "uuid" in the error message?